platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -665,6 +692,26 @@ - (id)mgl_jsonExpressionObject {
                     [expressionObject addObject:obj.mgl_jsonExpressionObject];
                 }];
                 [expressionObject addObject:self.operand.mgl_jsonExpressionObject];
+                return expressionObject;
+            } else if ([function isEqualToString:@"mgl_matchWithOptions:default:"]) {
+                NSMutableArray *expressionObject = [NSMutableArray arrayWithObjects:@"match", self.operand.mgl_jsonExpressionObject, nil];
+                NSArray *optionsArray = self.arguments[0].constantValue;
+                for (NSDictionary<id, NSExpression*> *options in optionsArray) {
```

Comment:
Unlike with coalescing, I think it could get quite inconvenient to have to specify all the cases for a match expression inside an aggregate expression. We can simplify the function name to `mgl_match:` by supposing a vararg method signature for the underlying Objective-C method:

```objc
- (id)mgl_match:(NSExpression *)firstCase, ...;
```

Then the arguments to this expression would be `firstCase`, `firstValue`, `secondCase`, `secondValue`, …, `defaultValue`.

(So far I’ve been taking cues from NSExpression’s built-in functions, not necessarily including all the nouns that an Objective-C method selector would, but still including prepositions.)

Reply:
Varargs are rare or nonexistent among NSExpression’s built-in functions, so when we implement the aftermarket version of this function, we should make it look like a control structure. Control structures like `TERTIARY()` don’t have selectors, so the developer wouldn’t expect to call `-expressionValueWithObject:context:` on them. Therefore, we could call the function `MGL_MATCH:` and not need to explicitly name the arguments or add colons for each one.